app/gui/src/tests.rs

```diff
@@ -82,13 +82,31 @@ fn span_tree_args() {
 
     let get_node = || graph.node(id).unwrap();
     let get_inputs = || NodeTrees::new(&get_node().info, executed_graph).unwrap().inputs;
-    let get_param =
-        |n| get_inputs().root_ref().leaf_iter().nth(n).and_then(|node| node.argument_info());
+    let get_param = |n| {
+        let inputs = get_inputs();
+        let mut args = inputs
+            .root_ref()
+            .leaf_iter()
+            .filter(|n| n.is_this() || n.is_argument() || n.is_expected_argument());
```

Comment:
This is pretty complex. Refactor it to several, well named variables descriibng what we get on every processing step, please.